tool/generator/lib/src/projection/type.dart

```diff
@@ -224,6 +224,37 @@ class TypeProjection {
     return TypeTuple(nativeType, dartType);
   }
 
+  TypeTuple unwrapWinRTDelegate() {
+    final delegateName = stripGenerics(
+        stripLeadingUnderscores(lastComponent(typeIdentifier.name)));
+
+    return TypeTuple('Pointer<NativeFunction<$delegateName>>',
+        'Pointer<NativeFunction<$delegateName>>');
+  }
+
+  TypeTuple unwrapReferenceType() {
+    if ((typeIdentifier.typeArg?.type?.isInterface ?? false) ||
+        typeIdentifier.typeArg?.baseType == BaseType.classTypeModifier) {
+      return const TypeTuple('Pointer<COMObject>', 'Pointer<COMObject>');
+    }
+
+    if (typeIdentifier.typeArg?.baseType == BaseType.simpleArrayType) {
+      // TODO: Handle PassArray and FillArray
```

Comment:
Since they are exposed as simple array types directly, these two are already handled by this:
https://github.com/timsneath/win32/blob/bb73f777c4e4712bceb5c72ed3e2ea903484d3ad/tool/generator/lib/src/projection/type.dart#L302

PassArray style is used in PropertyValue's `CreateXXArray` methods and currently, they are projected correctly.

FillArray style is tricky to test automatically because I couldn't find a method that uses this style other than IVectorView's `GetMany()`. Since IVectorView is a generic interface and GetMany method takes a generic array as a parameter, `classVariableTypeModifier` needs to be specified in `type.dart`. Without it, the generator throws an exception ("Type information missing for classVariableTypeModifier"). 

In order to check to see if GetMany method is generated correctly, I set `classVariableTypeModifier` to `IntPtr` in `type.dart` like this:
```dart
    ...
    if (isInterface ||
        typeIdentifier.baseType == BaseType.classTypeModifier ||
        typeIdentifier.baseType == BaseType.objectType) {
      return const TypeTuple('Pointer<COMObject>', 'Pointer<COMObject>');
    }

    if (typeIdentifier.baseType == BaseType.classVariableTypeModifier) {
      return const TypeTuple('IntPtr', 'IntPtr');
    }
    ...
```
... and generated IVectorView. Here is the generated GetMany method:
```dart
int GetMany(int startIndex, int valueSize, Pointer<IntPtr> value) {
    final retValuePtr = calloc<Uint32>();

    try {
      final hr = ptr.ref.vtable
              .elementAt(9)
              .cast<
                  Pointer<
                      NativeFunction<
                          HRESULT Function(
                              Pointer,
                              Uint32 startIndex,
                              Uint32 valueSize,
                              Pointer<IntPtr> value,
                              Pointer<Uint32>)>>>()
              .value
              .asFunction<
                  int Function(Pointer, int startIndex, int valueSize,
                      Pointer<IntPtr> value, Pointer<Uint32>)>()(
          ptr.ref.lpVtbl, startIndex, valueSize, value, retValuePtr);

      if (FAILED(hr)) throw WindowsException(hr);

      final retValue = retValuePtr.value;
      return retValue;
    } finally {
      free(retValuePtr);
    }
  }
```

The generated method is correct as the value parameter is wrapped with only one `Pointer`. You can see here the methods that I've generated manually in IVectorView to support retrieving COMObject and String arrays:
https://github.com/timsneath/win32/blob/fd55ddc5e1a1027a62eed567d0b229d4fab80235/lib/src/winrt/ivectorview.dart#L268-L348

Reply:
Gotcha. Removed.